nixos/modules/services/misc/ethminer.nix

```diff
@@ -0,0 +1,109 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ethminer;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.ethminer = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "enable ethminer ether mining.";
+      };
+
+      recheck = mkOption {
+        type = types.int;
+        default = 1000;
+        description = "time to recheck.";
+      };
+
+      toolkit = mkOption {
+        type = types.string;
+        default = "cuda";
+        description = "cuda or opencl.";
+      };
+
+      apiPort = mkOption {
+        type = types.int;
+        default = -3333;
+        description = "ethminer api port. minus sign puts api in read-only mode.";
+      };
+
+      wallet = mkOption {
+        type = types.string;
+        example = "0xa7E7030dbFf0Be4695a36C413C299Fd3469E83f3";
+        description = "ethereum wallet address.";
+      };
+
+      pool = mkOption {
+        type = types.string;
+        example = "eth-us-east1.nanopool.org";
+        description = "mining pool address.";
+      };
+
+      stratumPort = mkOption {
+        type = types.int;
+        default = 9999;
+        description = "stratum protocol tcp port.";
+      };
+
+      rig = mkOption {
+        type = types.string;
+        default = "mining-rig-name";
+        description = "mining rig name.";
+      };
+
+      notify = mkOption {
+        type = types.string;
+        example = "email%40example.org";
+        description = "url encoded email address to register with pool.";
+      };
+
+      power = mkOption {
+        type = types.int;
+        default = 115;
+        description = "miner max watt usage.";
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.ethminer = {
+      path = [ pkgs.cudatoolkit ];
+      description = "ethminer ethereum mining service";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        DynamicUser = true;
+        ExecStartPost = "+${pkgs.linuxPackages.nvidia_x11.bin}/bin/nvidia-smi -pl ${toString cfg.power}";
```

Comment:
please use `config.boot.kernelPackages` instead of `pkgs.linuxPackages`. This will automatically use a version built with the right kernel.

Reply:
done